### PR TITLE
Kubernetes Backend; Go Module Support

### LINKED ISSUE #414: compute: kubernetes support
Tracking issue for Kubernetes support. Currently, there is no support. We've discussed various ideas for integration.

### PR DESCRIPTION
**Pull request by adamstruck**
For setup see: https://github.com/ohsu-comp-bio/funnel/blob/kube/deployments/kubernetes/README.md

closes #414

TODO:
 - [x] More documentation
 - [x] Map disk request to volume in job template

### CODE REVIEW

**Comment by kellrott** [comment]
Tested against K8S cluster built using https://github.com/ohsu-comp-bio/openstack_k8s
When submitting a job, Funnel set it to a 'Queued' state and never updated. Looking into the logs found:

```
{"error":"creating job: jobs.batch is forbidden: User \"system:serviceaccount:default:default\" cannot create resource \"jobs\" in API group \"batch\" in the namespace \"default\"","level":"error","msg":"error submitting task to compute backend","ns":"server","taskID":"bpq39a19jgp5u6qfrh70","time":"2020-03-20T03:12:40Z"}
```

This setting probably needs to be added to the documentation.

File: .goreleaser.yml [hunk 1]
```diff
@@ -32,6 +32,16 @@ dockers:
     image_templates:
       - "ohsucompbio/funnel:{{.Version}}"
     dockerfile: ./docker/Dockerfile
+  - goos: linux
+    goarch: amd64
+    image_templates:
+      - "ohsucompbio/funnel-kube-dind:{{.Version}}"
+    dockerfile: ./deployments/kubernetes/dind/Dockerfile
```
Comment (adamstruck) [inline .goreleaser.yml]: This needs to be updated.

**Comment by adamstruck** [comment]
@kellrott I think this PR is ready for merge

**Comment by kellrott** [comment]
It's been tested on multiple K8S setups, so looks good.

**Pull request merged**